overlays: Add an overlay map

pelwell opened the pull request:
With a suitable firmware, an overlay map file can:
* allow a single overlay name to be mapped to platform-specific implementations,
* reject overlay names on some platforms,
* rename overlays, and
* mark overlays as deprecated,

all with helpful error messages.

These commits add a base map file (suitable for 4.19 if necessary), tidy up some old overlays, and add a remapping for vc4-kms-v3d on Pi 4.

Commit:
overlays: Add overlay_map

The overlay map permits platform-specific overlays, with deprecation
and renaming.

See: https://github.com/raspberrypi/linux/issues/3520

Signed-off-by: Phil Elwell <phil@raspberrypi.org>

Commit:
overlays: Add vc4-kms-v3d-pi4 to overlay_map

Signed-off-by: Phil Elwell <phil@raspberrypi.org>

Commit:
overlays: Formally rename/deprecate old overlays

Take advantage of the overlay_map to rename or deprecate some obsolete
overlays.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>

vianpl commented on arch/arm/boot/dts/overlays/overlay_map.dts:
```diff
@@ -108,4 +108,13 @@
+	};
+
+	vc4-kms-v3d-pi4 {
+		bcm2711;
```
I guess this could be implied by the property above. Although I see how it can make the implementation complicated.

pelwell replied:
Yes it is implied, but I don't really fancy having to search the whole map for string properties that match the current name every time an overlay is applied. However, I can add a test that confirms that mappings like this are consistent.

vianpl replied:
Fair enough :)

Commit:
Add upstream and upstream-pi4 to overlay_map

Because the upstream overlay applies vc4-kms-v3d, of which Pi 4 has its
own version, there also needs to be a Pi 4 version - vc4-kms-v3d-pi4.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>

popcornmix commented:
Looks good to me.

pelwell commented:
Thanks. It will need the firmware side pushing to be generally useful, but it's harmless until then,
